0000-catalog.md

```diff
@@ -0,0 +1,744 @@
+    string name = 3;
+    repeated PropertyValues properties = 4; 
+} 
+```
+Validation requirements:
+
+- If a catalog with catalog_id exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must belong to an 
+organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_update_catalog for the organization, 
+otherwise the transaction is invalid.
+
+If all requirements are met, the transaction will be accepted, the batch will be written 
+to a block, and the product will be created in state.
+
+The inputs for CatalogUpdateAction must include:
+
+Address of the Agent submitting the transaction
+Address of the Organization the Catalog is being created for
+Address of the Catalog to be updated
+
+The outputs for CatalogUpdateAction must include:
+
+Address of the Catalog to be updated
+
+### CatalogDeleteAction
+CatalogDeleteAction adds a new catalog to state. The transaction should be submitted 
+by an agent, which is identified by its signing key, acting on behalf of the 
+organization that corresponds to the owner in the delete transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+```
+message CatalogDeleteAction { 
+    // GS1 Company Prefix from owner + catalog_id are use as 
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+} 
+```
+Validation requirements:
+
+- If a catalog with catalog_id exists the transaction is valid, otherwise it's invalid.
+- The signer of the transaction must be an agent in the Pike state and must belong to 
+an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_delete_catalog for the organization, otherwise 
+the transaction is invalid.
+
+If all requirements are met, the transaction will be accepted, the batch will be 
+written to a block, and the catalog will be created in state.
+
+The inputs for CatalogDeleteAction must include:
+
+Address of the Agent submitting the transaction
+Address of the Organization the Catalog is being created for
+Address of the Catalog to be deleted
+
+The outputs for CatalogDeleteAction must include:
+
+Address of the Catalog to be deleted
+
+## Catalog_Product Actions
+
+### CatalogProductCreateAction
+The CatalogProductCreateAction adds a new catalog_product to state. The transaction should 
+be submitted by an agent, which is identified by its signing key, acting on behalf of the 
+organization that corresponds to the owner in the create transaction. (Organizations and 
+agents are defined by the Pike smart contract.)
+```
+message CatalogProductCreateAction { 
+    // catalog_id and product_id are used in deriving the state address
+    string catalog_id = 1
+    string product_id = 2 // Reference from Grid Product 
+    // schema defined additional fields
+    repeated PropertyValues properties = 4;
+} 
+```
+Validation requirements:
```

Comment:
Does there need to be a product in normal product state that correlates to this CatalogProduct?

Reply:
Changed to catalog_product_id ✅